From 27d1b5d3804d484ce0338ffd531f336cac9eb751 Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Fri, 22 Sep 2017 02:14:25 +0800
Subject: [PATCH 3/3] enable shared linking

Signed-off-by: Xuefer <xuefer@gmail.com>
---
 bindings/python/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/python/Makefile.am b/bindings/python/Makefile.am
index c909ac5..f1b677c 100644
--- a/bindings/python/Makefile.am
+++ b/bindings/python/Makefile.am
@@ -33,7 +33,7 @@ EXTRA_DIST = \
 if ENABLE_PYTHON_BINDING
 
 all-local:
-	$(PYTHON) $(srcdir)/setup.py build
+	LDCXXSHARED="$(CXX) -shared" $(PYTHON) $(srcdir)/setup.py build
 
 install-exec-local:
 	$(PYTHON) $(srcdir)/setup.py install @PYTHON_INSTALL_PARAMS@
-- 
2.10.2

