From 87e5d109f0a7eaf22fa9ba115f31871722868092 Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Fri, 22 Sep 2017 21:00:22 +0800
Subject: [PATCH 4/4] delay CXXLIBS to avoid test link error

Signed-off-by: Xuefer <xuefer@gmail.com>
---
 configure.ac | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6681a39..7094a10 100644
--- a/configure.ac
+++ b/configure.ac
@@ -133,7 +133,7 @@ AS_IF([test -z "$BOOST_RANDOM_LIB"],
 
 CPPFLAGS="$BOOST_CPPFLAGS $CPPFLAGS"
 LDFLAGS="$BOOST_LDFLAGS $LDFLAGS"
-LIBS="$BOOST_CHRONO_LIB $BOOST_RANDOM_LIB $LIBS"
+CXXLIBS="$BOOST_CHRONO_LIB $BOOST_RANDOM_LIB"
 
 ###############################################################################
 # Checking for functions and other stuffs
@@ -515,6 +515,8 @@ AS_CASE(["$ARG_WITH_LIBICONV"],
    AC_MSG_ERROR([Unknown option "$ARG_WITH_LIBICONV". Use either "yes" or "no".])]
 )
 
+LIBS="$LIBS $CXXLIBS"
+
 ###############################################################################
 # Setting conditional variables for Makefiles
 ###############################################################################
-- 
2.10.2

