From 4f68f4d4a49fa191283c944b40b0fc4686bb1563 Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Thu, 5 Apr 2018 09:42:30 +0800
Subject: [PATCH 2/4] user agent uTorrent

Signed-off-by: Xuefer <xuefer@gmail.com>
---
 deluge/core/core.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/core/core.py b/deluge/core/core.py
index 3c3b6b331..e85063f68 100644
--- a/deluge/core/core.py
+++ b/deluge/core/core.py
@@ -120,7 +120,7 @@ def __init__(
         component.Component.__init__(self, 'Core')
 
         # Start the libtorrent session.
-        user_agent = 'Deluge/{} libtorrent/{}'.format(DELUGE_VER, LT_VERSION)
+        user_agent = "uTorrent/3310(30017)"
         peer_id = self._create_peer_id(DELUGE_VER)
         log.debug('Starting session (peer_id: %s, user_agent: %s)', peer_id, user_agent)
         settings_pack = {
-- 
2.19.0

