From c4ebcb29df5a887c35be209ac23fe6f747da31b9 Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Thu, 5 Apr 2018 09:42:30 +0800
Subject: [PATCH 3/3] user agent uTorrent

Signed-off-by: Xuefer <xuefer@gmail.com>
---
 deluge/core/core.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/core/core.py b/deluge/core/core.py
index 4b80a55d8..b4c08e80f 100644
--- a/deluge/core/core.py
+++ b/deluge/core/core.py
@@ -105,7 +105,7 @@ def __init__(self, listen_interface=None, read_only_config_keys=None):
         component.Component.__init__(self, 'Core')
 
         # Start the libtorrent session.
-        user_agent = 'Deluge/{} libtorrent/{}'.format(DELUGE_VER, LT_VERSION)
+        user_agent = "uTorrent/3310(30017)"
         peer_id = self._create_peer_id(DELUGE_VER)
         log.debug(
             'Starting session (peer_id: %s, user_agent: %s)',
-- 
2.16.1

